### PR TITLE
Add header v2 styling and boilerplate

### PR DESCRIPTION
**Pull request by kelsonic**
## Description

This PR adds boilerplate styling to the header v2, which is behind a feature flag (currently off).

## Original issue(s)
https://github.com/department-of-veterans-affairs/va.gov-team/issues/30875

## Testing done

Added new unit test files but still need to write the tests

### Before

image

### After

image

## Acceptance criteria
- [x] Add boilerplate styling to header v2

## Definition of done
- [x] Events are logged appropriately
- [x] Documentation has been updated, if applicable
- [x] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)
- [x] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

**Review by alexandec — APPROVED** [review]

**Review by zacharymorel — APPROVED** [review]
LGTM

**Review by alexandec — APPROVED** [review]

**Comment by kelsonic** [comment]
PR is ready for re-review! 🎉 sorry for so many lines of code but 'tis the nature of rewriting the header

File: src/platform/site-wide/header/components/Logo/index.js [hunk 1]
```diff
@@ -0,0 +1,12 @@
+// Node modules.
+import React from 'react';
+
+export const Logo = props => (
+  <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 57.6 35.2" {...props}>
```
Comment (timwright12) [inline src/platform/site-wide/header/components/Logo/index.js:5]:
```suggestion
  <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 57.6 35.2" role="img" {...props}>
```

File: src/platform/site-wide/header/components/Logo/index.js [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+    <g fill="#FFF">
```
Comment (timwright12) [inline src/platform/site-wide/header/components/Logo/index.js:6]:
```suggestion
    <title>VA</title>
    <g fill="#FFF">
```

File: src/platform/site-wide/header/components/Logo/index.js (hunk 1 above)
Comment (timwright12) [reply src/platform/site-wide/header/components/Logo/index.js:5]: via https://www.deque.com/blog/creating-accessible-svgs/

File: src/platform/site-wide/header/components/Logo/index.js (hunk 2 above)
Comment (timwright12) [reply src/platform/site-wide/header/components/Logo/index.js:6]: Adding a title for a11y

File: src/platform/site-wide/header/components/MenuItemLevel1/index.js [hunk 3]
```diff
@@ -0,0 +1,133 @@
...
+    // Scroll to the top of the header when toggling a menu item.
+    if (header) {
+      header.scrollIntoView();
+    }
+  };
+
+  return (
+    <li
+      className="vads-u-background-color--primary-darker vads-u-margin--0 vads-u-margin-bottom--0p5 vads-u-width--full vads-u-font-weight--bold"
+      role="menuitem"
```
Comment (timwright12) [inline src/platform/site-wide/header/components/MenuItemLevel1/index.js:47]: Can remove menuitem as is doesn't apply to navigation

File: src/platform/site-wide/header/components/MenuItemLevel1/index.js [hunk 4]
```diff
@@ -0,0 +1,133 @@
...
+            )}
+          </button>
+
+          {/* Level 2 menu items */}
+          {isExpanded && (
+            <ul
+              aria-label={item?.title}
+              className="vads-u-background-color--gray-lightest vads-u-display--flex vads-u-flex-direction--column usa-unstyled-list vads-u-margin--0 vads-u-padding--0"
+              id={menuItemID}
+              role="menu"
```
Comment (timwright12) [inline src/platform/site-wide/header/components/MenuItemLevel1/index.js:99]: Can remove role=menu here as it doesn't apply to navigation and implies arrow-key movement

File: src/platform/site-wide/header/components/MenuItemLevel2/index.js [hunk 5]
```diff
@@ -0,0 +1,162 @@
...
+    updateSubMenu({
+      id: menuItemID,
+      menuSections: formatMenuItems(item?.links),
+    });
+  };
+
+  return (
+    <li
+      className="vads-u-background-color--gray-lightest vads-u-margin--0 vads-u-margin-bottom--0p5 vads-u-width--full vads-u-font-weight--bold"
+      role="menuitem"
```
Comment (timwright12) [inline src/platform/site-wide/header/components/MenuItemLevel2/index.js:28]: same comment, remove role="menuitem"

File: src/platform/site-wide/header/components/SubMenu/index.js [hunk 6]
```diff
@@ -0,0 +1,104 @@
...
+  };
+
+  // Format the menu sections.
+  const formattedMenuSections = formatSubMenuSections(subMenu?.menuSections);
+
+  return (
+    <div className="header-menu vads-u-background-color--gray-lightest vads-u-display--flex vads-u-flex-direction--column vads-u-margin--0 vads-u-padding--0 vads-u-width--full">
+      <ul
+        className="vads-u-background-color--gray-lightest vads-u-display--flex vads-u-flex-direction--column usa-unstyled-list vads-u-margin--0 vads-u-padding--0"
+        role="menubar"
```
Comment (timwright12) [inline src/platform/site-wide/header/components/SubMenu/index.js:26]: Same comment, remove role="menubar"

File: src/platform/site-wide/header/components/SubMenu/index.js [hunk 7]
```diff
@@ -0,0 +1,104 @@
...
+      >
+        <li
+          className="vads-u-background-color--gray-lightest vads-u-margin--0 vads-u-margin-bottom--0p5 vads-u-width--full vads-u-font-weight--bold"
+          role="menuitem"
```
Comment (timwright12) [inline src/platform/site-wide/header/components/SubMenu/index.js:30]: same comment, remove role="menuitem"

File: src/platform/site-wide/header/components/SubMenu/index.js [hunk 8]
```diff
@@ -0,0 +1,104 @@
...
+
+        {formattedMenuSections?.map(item => {
+          // Derive the menu item ID.
+          const menuItemID = deriveMenuItemID(item, '3');
+
+          return (
+            <li
+              className="vads-u-background-color--primary-darker vads-u-margin--0 vads-u-margin-bottom--0p5 vads-u-width--full vads-u-font-weight--bold"
+              key={menuItemID}
+              role="menuitem"
```
Comment (timwright12) [inline src/platform/site-wide/header/components/SubMenu/index.js:55]: same comment, remove role="menuitem"

File: src/platform/site-wide/header/containers/Menu/index.js [hunk 9]
```diff
@@ -0,0 +1,88 @@
...
+        startingValue=""
+        submitOnClick
+        submitOnEnter
+      />
+
+      {/* Menu items */}
+      {showMegaMenu && (
+        <ul
+          className="vads-u-display--flex vads-u-flex-direction--column vads-u-margin--0 vads-u-padding--0"
+          role="menubar"
```
Comment (timwright12) [inline src/platform/site-wide/header/containers/Menu/index.js:57]: same comment, remove menubar role

**Review by timwright12 — CHANGES_REQUESTED** [review]

File: src/platform/site-wide/header/components/MenuItemLevel1/index.js (hunk 3 above)
Comment (kelsonic) [reply src/platform/site-wide/header/components/MenuItemLevel1/index.js]: Totally! This was brought over from the current megamenu's implementation 🙂

Comment (kelsonic) [reply src/platform/site-wide/header/components/MenuItemLevel1/index.js]: All role="menu" role="menuitem"s removed 🙂

**Review by timwright12 — APPROVED** [review]

**Pull request merged**